src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+ */
+void DemoEffect_JewelSparkle(DemoEffect* this, GlobalContext* globalCtx, s32 spawnerCount) {
+    Vec3f velocity;
+    Vec3f accel;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+    Color_RGB8* sparkleColors;
+    s32 i;
+
+    i = 0;
+
+    velocity.y = 0.0f;
+
+    accel.x = 0.0f;
+    accel.y = -0.1f;
+    accel.z = 0.0f;
+
+    sparkleColors = jewelSparkleColors[this->jewel.type - Demo_Effect_Jewel_Kokiri];
+
+    primColor.r = (sparkleColors + 0)->r;
+    primColor.g = (sparkleColors + 0)->g;
+    primColor.b = (sparkleColors + 0)->b;
+    envColor.r = (sparkleColors + 1)->r;
+    envColor.g = (sparkleColors + 1)->g;
+    envColor.b = (sparkleColors + 1)->b;
+    primColor.a = 0;
+
+    while (i < spawnerCount) {
+        velocity.x = (Math_Rand_ZeroOne() - 0.5f) * 1.5f;
+        velocity.z = (Math_Rand_ZeroOne() - 0.5f) * 1.5f;
+
+        EffectSsKiraKira_SpawnDispersed(globalCtx, &this->actor.posRot.pos, &velocity, &accel, &primColor, &envColor,
+                                        3000, 16);
+
+        i++;
+    }
+}
+
+/**
+ * Plays Jewel sound effects.
+ * The sfxJewelId global variable is used to ensure only one Jewel Actor is playing SFX when all are spawned.
+ */
+void DemoEffect_PlayJewelSfx(DemoEffect* this, GlobalContext* globalCtx) {
+    if (!DemoEffect_CheckCsAction(this, globalCtx, 1)) {
+        if (this->actor.params == sfxJewelId[0]) {
+            func_8002F974(&this->actor, NA_SE_EV_SPIRIT_STONE - SFX_FLAG);
+        } else if (!sfxJewelId[0]) {
+            sfxJewelId[0] = this->actor.params;
+            func_8002F974(&this->actor, NA_SE_EV_SPIRIT_STONE - SFX_FLAG);
+        }
+    }
+}
+
+/**
+ * Update Function for the Jewel Actor that is run when Link is an adult.
+ * This rotates the Jewel and updates a timer that is used to scroll Jewel textures.
+ * There is a call SetJewelColor that does nothing since 1.0f is passed.
+ * If a value of less than 1.0f were passed to SetJewelColor, then it would appear to drain the Jewel's color.
+ * This can be seen in preprelease screenshots.
+ */
+void DemoEffect_UpdateJewelAdult(DemoEffect* this, GlobalContext* globalCtx) {
+    this->jewel.timer++;
+    this->actor.shape.rot.y += 1024;
+    DemoEffect_PlayJewelSfx(this, globalCtx);
+    DemoEffect_SetJewelColor(this, 1.0f);
+}
+
+/**
+ * Update Function for the Jewel Actor that is run when Link is a child.
+ * This rotates the Jewel and updates a timer that is used to scroll Jewel textures.
+ * This also updates the Jewel's position based on different cutscenes.
+ */
+void DemoEffect_UpdateJewelChild(DemoEffect* this, GlobalContext* globalCtx) {
+    s32 hasCmdAction;
+    Actor* thisx = &this->actor;
+
+    this->jewel.timer++;
+
+    if (globalCtx->csCtx.state && globalCtx->csCtx.npcActions[this->csActionId]) {
+        switch (globalCtx->csCtx.npcActions[this->csActionId]->action) {
+            case 3:
+                if (gSaveContext.eventChkInf[4] & 2048) {
```

Comment:
```suggestion
                if (gSaveContext.eventChkInf[4] & 0x800) {
```
Bitflags are definitely in hex. Possibly should be `(1 << 11)`.

Reply:
Yeah I like that better.